src/cleanup.ts

```diff
@@ -1,4 +1,4 @@
-let cleanupCallbacks = []
+let cleanupCallbacks: (() => Promise<void>)[] = []
```

Comment:
I think the return type of callbacks should be `Promise<void>|void` in case the callback is not async

Reply:
I had done it this way, as the callbacks are always called with `await` - Does it still make sense to be `Promise<void> | void` still?

Reply:
`await` can still work if the code is synchronous you can see an example in [the documentation for the await operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await).
so it makes sense to use `await` in this case because it may or may not be a promise

Reply:
@merodiro Thanks! Fixing that now...